core/Pos/Binary/Core/Types.hs

```diff
@@ -61,3 +61,32 @@ instance Bi T.SharedSeed where
 instance Bi T.ChainDifficulty where
     get = label "ChainDifficulty" $ T.ChainDifficulty . getUnsignedVarInt <$> get
     put (T.ChainDifficulty c) = put (UnsignedVarInt c)
+
+instance Bi T.BlockVersionData where
+    get = label "BlockVersionData" $ do
```

Comment:
`deriving` is also bad because it doesn't have `label`. Maybe we can use `Generics` to implement binary serialization, it was proposed several times earlier, this could be ok (if it doesn't decrease performance).